### PR TITLE
choose encoding and create --force mode

### PR DESCRIPTION
**Pull request by narsonalin**
This pull request enables the change of encoding used to read source files (which is useful in case your source files include accentuated characters, not encoding in utf8).

Also added the possibility to create a modifications history in each subroutines as a note type (@history).

Finally, add the possibility to print the faulty line in case --debug is activated. And interpret fatal error as warnings so that one can get the whole list of faulty lines in a file (useful for f77)

### CODE REVIEW

File: ford/reader.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ class FortranReader(object):
 
     def __init__(self,filename,docmark='!',predocmark='',docmark_alt='',
                  predocmark_alt='',fixed=False,length_limit=True,
-                 preprocessor=None,macros=[],inc_dirs=[]):
+                 preprocessor=None,macros=[],inc_dirs=[], encoding='utf-8'):
```
Comment (zbeekman) [inline ford/reader.py]: Why is this hard-coded to 'utf-8' and elsewhere it is set dynamically?

File: ford/reader.py [hunk 2]
```diff
@@ -291,7 +290,7 @@ def __next__(self):   # Python 3
         else:
             tmp = self.docbuffer.pop(0)
             if tmp != "!"+self.docmark:
-                self.prevdoc = True;
+                self.prevdoc = True
```
Comment (zbeekman) [inline ford/reader.py]: Why?

File: ford/sourceform.py [hunk 3]
```diff
@@ -778,15 +797,23 @@ def __init__(self,source,first_line,parent=None,inherited_permission=None,
...
                     # has been misidentified as a function call
-                    #~ raise Exception("Found procedure call in {}".format(type(self).__name__[7:].upper()))
+                    #if self.settings["dbg"]:
+                    #    print(line)
+                    #    print("Found procedure call in {}".format(type(self).__name__[7:].upper()))
+                    #else:
+                    #if self.settings["force"]:
+                    #    continue
+                    #else:
+                    #    raise Exception("Found procedure call in {}".format(type(self).__name__[7:].upper()))
```
Comment (zbeekman) [inline ford/sourceform.py]: Why is this added but commented out?

**Comment by zbeekman** [comment]
The last change (continuing processing and printing faulty lines) looks really great!

Could you please include a screenshot of how the new `@history` note renders?

**Review by ZedThree — APPROVED** [review]
